app/controllers/goals_controller.rb

```diff
@@ -1,20 +1,28 @@
 class GoalsController < ApplicationController
 
   def index
-    @sweat = Category.find(1)
-    @sleep = Category.find(2)
-    @personal = Category.find(3)
-    @nutrition = Category.find(4)
-    
-    @sweat_progress = Goal.community_progress(current_week.id, 1)
-    @sleep_progress = Goal.community_progress(current_week.id, 2)
-    @personal_progress = Goal.community_progress(current_week.id, 3)
-    @nutrition_progress = Goal.community_progress(current_week.id, 4)
+    @categories = Category.all
+  end
+
+  def increment
+    @goal = current_user.goals.find(params[:id])
+    new_count = @goal.progress_count + 1
+
+
+    if @goal.update(progress_count: new_count)
+      render json: {progress_count: @goal.progress_count, total_goal_count: @goal.total_goal_count}.to_json
```

Comment:
nice 👍 

be sure to account for what happens if the goal does NOT update for whatever reason. what happens from a user standpoint? how do they know their action didn't work?